cpp/cmake/thirdparty/CUDF_GetcuCollections.cmake

```diff
@@ -0,0 +1,38 @@
+#=============================================================================
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+function(find_and_configure_cucollections)
+
+    if(TARGET cuCollections::cuco)
+        return()
+    endif()
+
+    # Find or install cuCollections
+    CPMFindPackage(NAME   cuco
```

Comment:
Currently `cudf` depends on libcudacxx 1.4, and thrust 1.12, while cuCollections depends on libcudacxx 1.3 and thrust 1.10. 

At a minimum we should comment that `CUDF_GetcuCollections.cmake` must be called after the calls for Thrust and libcudacxx to make sure we get the higher versions required. Second we should do the following to make sure we never introduce a secondary copy of Thrust of libcudacxx.

```cmake
set(CPM_LOCAL_PACKAGES_ONLY ON)
CPMFindPackage(NAME   cuco .... )
```

Reply:
I tried this and got an error that i understand to mean that cuco wasn't found. Meaning that this might be preventing cuco itself to be downloaded.

Reply:
What was the specific error?